libelement/src/interpreter.cpp

```diff
@@ -201,7 +202,21 @@ element_result element_instruction_get_size(element_instruction* instruction, si
     return ELEMENT_OK;
 }
 
-element_result element_instruction_is_constant(element_instruction* instruction, bool* constant)
+element_result element_instruction_get_inputs_size(const element_instruction* instruction, size_t* size)
+{
+    if (!instruction || !instruction->instruction)
+        return ELEMENT_ERROR_API_INSTRUCTION_IS_NULL;
+
+    if (!size)
+        return ELEMENT_ERROR_API_OUTPUT_IS_NULL;
+
+    *size = 0;
+    for (const auto& d : instruction->instruction->dependents())
+        *size += d->get_size();
+    return ELEMENT_OK;
+}
+
```

Comment:
could be worth mentioning the difference between this and `element_instruction_get_size` is that the other one doesn't return `0` for instructions without dependents 🤷 (I had to double check what it was doing and why it wasn't the same :b)